### PR TITLE
Update instance routing to use org/instance

### PR DESCRIPTION
**Pull request by fmoor**
depends on https://github.com/edgedb/nebula/pull/14

### CODE REVIEW

File: src/cloud/ops.rs [hunk 1]
```diff
@@ -141,43 +148,43 @@ pub async fn create_cloud_instance(
...
-    let instance = wait_instance_create(instance, client, false).await?;
-    write_credentials(&cred_path, instance).await?;
+    wait_instance_create(instance, client, false).await?;
     Ok(())
 }
 
+pub fn split_cloud_instance_name(name: &str) -> anyhow::Result<(String, String)> {
+    let mut splitter = name.splitn(2, '/');
+    match splitter.next() {
+        None => anyhow::bail!("unreachable"),
```
Comment (tailhook) [inline src/cloud/ops.rs:162]:
```suggestion
        None => unreachable!(),
```

File: src/portable/options.rs [hunk 2]
```diff
@@ -466,10 +454,35 @@ impl fmt::Display for StartConf {
...
-                (regex: ^[a-zA-Z_][a-zA-Z_0-9]*$)".into())
+        regex: ^[a-zA-Z_][a-zA-Z_0-9]*$".into())
+}
+
+pub fn instance_or_cloud_name_opt(name: &str) -> Result<(), String> {
+    if is_valid_instance_name(&name) {
+        return Ok(())
+    }
+    if !name.contains("/") {
+        // todo: should this also mention the format for cloud instance names?
```
Comment (tailhook) [inline src/portable/options.rs:469]: Sure.

**Review by tailhook — APPROVED** [review]

File: src/portable/options.rs (hunk 2 above)
Comment (fmoor) [reply src/portable/options.rs:469]: I'm not sure how releases work in this project. Are we worried about hinting at cloud instance names in a pre 2.0 release of the CLI?

Comment (fmoor) [reply src/portable/options.rs:469]: or at least before the cloud is actually available?

Comment (elprans) [reply src/portable/options.rs:469]: The upcoming cloud is not a secret, so it's fine to mention.

File: src/cloud/ops.rs [hunk 3]
```diff
@@ -141,43 +148,43 @@ pub async fn create_cloud_instance(
...
-    if cred_path.exists() {
-        anyhow::bail!("File {} exists; abort.", cred_path.display());
-    }
+    let url = format!("orgs/{}/instances", instance.org_slug);
     let instance: CloudInstance = client
-        .post("instances/", serde_json::to_value(instance)?)
+        .post(url, serde_json::to_value(instance)?)
         .await?;
-    let instance = wait_instance_create(instance, client, false).await?;
-    write_credentials(&cred_path, instance).await?;
```
Comment (nsidnev) [inline src/cloud/ops.rs:152]: I think removing this line is a regression, no? I see one for linking the cloud instance, but nothing for creating it.

File: src/options.rs [hunk 4]
```diff
@@ -629,7 +631,15 @@ pub fn conn_params(tmp: &ConnectionOptions) -> anyhow::Result<Builder> {
         task::block_on(bld.read_dsn(dsn))?;
         bld.read_extra_env_vars()?;
     } else if let Some(instance) = &tmp.instance {
-        task::block_on(bld.read_instance(instance))?;
+        if instance.contains("/") {
```
Comment (nsidnev) [inline src/options.rs:634]: Shouldn't this PR also change the instance name parsing part of the Rust client before merging? Right now, for the name with format like `org_name/instance_name` CLI will complain about invalid characters for the name.

I think the problem is here: https://github.com/edgedb/edgedb-rust/blob/6b385b3c9f8c548fddc61cd53c1182c13c92ded7/edgedb-client/src/builder.rs#L296

Comment (elprans) [reply src/options.rs:634]: Yes, the client validation should be updated too. `<org>/<instance>` should be automatically treated as a Cloud instance.

File: src/cloud/ops.rs (hunk 3 above)
Comment (fmoor) [reply src/cloud/ops.rs:154]: The other one should be removed too. Thanks for catching that.

**Review by elprans — APPROVED** [review]

**Pull request merged**